pillar/vault/roles/xpro.sls

```diff
@@ -0,0 +1,26 @@
+vault:
+  roles:
+    xpro-app:
+      backend: postgresql-xpro
+      name: app
+      options:
+        {% raw %}
+        sql: >-
+          CREATE USER "{{name}}" WITH PASSWORD '{{password}}' VALID UNTIL '{{expiration}}';
+          GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "{{name}}";
+          GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO "{{name}}";
+          ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL PRIVILEGES ON TABLES TO "{{name}}";
+          ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL PRIVILEGES ON SEQUENCES TO "{{name}}";
+        {% endraw %}
```

Comment:
You should copy the creation and revocation SQL from the `apps.sls` file for how the PostGres data is structured there. You could probably even just add an additional parameter to the loop that is already in there.